### DISCUSSION TITLE
Error upgrading from 9.3.2 to 9.7.2

### DISCUSSION THREAD

**Discussion by jncraig** [dnnsoftware/Dnn.Platform#4125] [General]
## Description of bug
Upgrade from 9.3.2 to 9.7.2 (part of an upgrade from 7.4.2 via 8.0.4, 9.1.1 to 9.3.2)

## Steps to reproduce
Copy files from 9.7.2 upgrade package to 9.3.2 web directory an browse to site to start the upgrade.

On screen report are:

Upgrading DNN
Current Assembly Version: 09.07.02
Current Database Version: 09.03.02

Upgrade Status Report
00:00:00.075 - Upgrading to Version: 09.07.02
00:00:00.104 -   Executing Script:09.04.00.SqlDataProvider Error! (see 09.04.00.log.resources for more information)
00:00:00.302 -   Executing Script:09.04.01.SqlDataProvider Success
00:00:00.312 -   Executing Script:09.04.02.SqlDataProvider Success
00:00:00.323 -   Executing Script:09.04.03.SqlDataProvider Success
00:00:00.346 -   Executing Script:09.04.04.SqlDataProvider Success
00:00:00.359 -   Executing Script:09.05.00.SqlDataProvider Success
00:00:00.427 -   Executing Script:09.05.01.SqlDataProvider Success
00:00:00.457 -   Executing Script:09.06.00.SqlDataProvider Success
00:00:00.501 -   Executing Script:09.06.01.SqlDataProvider Success
00:00:00.511 -   Executing Script:09.06.02.SqlDataProvider Success
00:00:00.779 -   Executing Script:09.07.00.SqlDataProvider Success
00:00:00.789 -   Executing Script:09.07.01.SqlDataProvider Success
00:00:00.810 -   Executing Script:09.07.02.SqlDataProvider Success
00:00:00.825 -   Application Upgrades : 09.04.00 Success
00:00:00.826 -   Application Upgrades : 09.04.01 Success
00:00:00.836 -   Application Upgrades : 09.04.02 Success
00:00:00.838 -   Application Upgrades : 09.04.03 Success
00:00:00.840 -   Application Upgrades : 09.04.04 Success
00:00:00.842 -   Application Upgrades : 09.05.00 Success
00:00:00.844 -   Application Upgrades : 09.05.01 Success
00:00:00.845 -   Application Upgrades : 09.06.00 Success
00:00:00.865 -   Application Upgrades : 09.06.01 Success
00:00:00.867 -   Application Upgrades : 09.06.02 Success
00:00:00.868 -   Application Upgrades : 09.07.00 Success
00:00:00.870 -   Application Upgrades : 09.07.01 Success
00:00:00.872 -   Application Upgrades : 09.07.02 Success
00:00:00.875 -   Cleaning Up Files: 09.04.00 Success
00:00:00.910 -   Cleaning Up Files: 09.04.01 Success
00:00:00.911 -   Cleaning Up Files: 09.04.02 Success
00:00:00.913 -   Cleaning Up Files: 09.04.03 Success
00:00:00.915 -   Cleaning Up Files: 09.04.04 Success
00:00:00.916 -   Cleaning Up Files: 09.05.00 Success
00:00:00.918 -   Cleaning Up Files: 09.05.01 Success
00:00:00.920 -   Cleaning Up Files: 09.06.00 Success
00:00:00.921 -   Cleaning Up Files: 09.06.01 Success
00:00:00.923 -   Cleaning Up Files: 09.06.02 Success
00:00:00.925 -   Cleaning Up Files: 09.07.00 Success
00:00:00.927 -   Cleaning Up Files: 09.07.01 Success
00:00:00.928 -   Cleaning Up Files: 09.07.02 Success
00:00:00.932 -   Updating Config Files: 09.04.00 Success
00:00:00.961 -   Updating Config Files: 09.04.01 Success
00:00:00.967 -   Updating Config Files: 09.04.02 Success
00:00:00.974 -   Updating Config Files: 09.04.03 Success
00:00:00.983 -   Updating Config Files: 09.04.04 Success
00:00:00.989 -   Updating Config Files: 09.05.00 Success
00:00:01.011 -   Updating Config Files: 09.05.01 Success
00:00:01.021 -   Updating Config Files: 09.06.00 Success
00:00:01.035 -   Updating Config Files: 09.06.01 Success
00:00:01.046 -   Updating Config Files: 09.06.02 Success
00:00:01.058 -   Updating Config Files: 09.07.00 Success
00:00:01.068 -   Updating Config Files: 09.07.01 Success
00:00:01.074 -   Updating Config Files: 09.07.02 Success
00:00:01.113 - Performing General Upgrades
00:00:01.519 -   Installing Package File DNNCE_Web.Deprecated_09.07.02_Install:  Success
00:00:01.750 -   Installing Package File DNNCE_Website.Deprecated_09.07.02_Install:  Success
00:00:01.893 -   Installing Package File Dnn.EditBar.UI_09.07.02_Install:  Success
00:00:02.298 -   Installing Package File DDRMenu_09.07.02_Install:  Error!
00:00:02.569 -   Installing Package File Dnn.PersonaBar.UI_09.07.02_Install:  Success
00:00:03.131 -   Installing Package File DNNCE_AzureConnector_09.07.02_Install:  Success
00:00:03.441 -   Installing Package File DNNCE_Console_09.07.02_Install:  Error!
00:00:03.625 -   Installing Package File DNNCE_CoreMessaging_09.07.02_Install:  Error!
00:00:03.800 -   Installing Package File DNNCE_HtmlEditorManager_09.07.02_Install:  Error!
00:00:03.970 -   Installing Package File DNNCE_Journal_09.07.02_Install:  Error!
00:00:04.144 -   Installing Package File DNNCE_MemberDirectory_09.07.02_Install:  Error!
00:00:04.317 -   Installing Package File DNNCE_ModuleCreator_09.07.02_Install:  Error!
00:00:04.491 -   Installing Package File DNNCE_RazorHost_09.07.02_Install:  Error!
00:00:04.738 -   Installing Package File DNNCE_SiteExportImport_09.07.02_Install:  Success
00:00:04.949 -   Installing Package File DNNCE_SocialGroups_09.07.02_Install:  Error!
00:00:05.119 -   Installing Package File GoogleAnalyticsConnector_09.07.02_Install:  Success
00:00:05.274 -   Installing Package File Telerik_09.07.02_Install:  Success
00:00:06.620 -   Installing Package File jQuery_03.05.01_Install:  Success
00:00:06.794 -   Installing Package File Skin_Xcillion_09.07.02_Install:  Success
00:00:07.194 -   Installing Package File Knockout_03.03.00_Install:  Success
00:00:07.356 -   Installing Package File Selectize_00.12.06_Install:  Success
00:00:07.510 -   Installing Package File AspNetClientCapabilityProvider_09.07.02_Install:  Success
00:00:07.687 -   Installing Package File DNN_SimpleWebFarmCachingProvider_09.07.02_Install:  Success
00:00:07.842 -   Installing Package File DNNCE_FolderProviders_09.07.02_Install:  Success
00:00:08.070 -   Installing Package File DNNConnect.CKEditorProvider_01.01.00_Install:  Success
00:00:11.474 -   Installing Package File Dnn.PersonaBar.Extensions_09.07.02_Install:  Success
00:00:12.826 -   Installing Package File DNNCE_DigitalAssetsManagement_09.07.02_Install:  Error!
00:00:13.015 -   Installing Package File DNNCE_HTML_09.07.02_Install:  Error!
00:00:13.238 -   Installing Package File jQueryMigrate_03.02.00_Install:  Success
Upgrade Complete

## Current behavior
See above for errors.

## Expected behavior
Nice clean install

## Screenshots
If applicable, provide screenshots to help explain the bug.

## Error information
Provide any error information (console errors, error logs, etc.) related to this bug.
[2020.09.24.log.zip](https://github.com/dnnsoftware/Dnn.Platform/files/5278337/2020.09.24.log.zip)

## Additional context
Provide any additional context that may be helpful in understanding and/or resolving the bug.

## Affected version

* [ ] 10.00.00 alpha build
* [ ] 09.07.02 release candidate
* [ ] 09.07.01 latest supported release

## Affected browser

* [ ] Chrome
* [ ] Firefox
* [ ] Safari
* [ ] Internet Explorer 11
* [ ] Microsoft Edge (Classic)
* [ ] Microsoft Edge Chromium

**Comment by bdukes** [comment]
@jncraig what's in `Portals/_default/Logs/09.04.00.log.resources`?

**Comment by jncraig** [comment]
This is from installer resources file:  InstallerLog20200924.resources   I dont see any09.04.00.log.resources file.

09/24/2020 18:13:22 [INFO] DotNetNuke.Services.Upgrade.Upgrade Start UpgradeVersion:E:\Websites\sbhs65\Website\Providers\DataProviders\SqlDataProvider\09.04.00.SqlDataProvider
09/24/2020 18:13:22 [INFO] DotNetNuke.Services.Upgrade.Upgrade Start ExecuteScript:E:\Websites\sbhs65\Website\Providers\DataProviders\SqlDataProvider\09.04.00.SqlDataProvider
09/24/2020 18:13:22 [ERROR] DotNetNuke.Services.Upgrade.Upgrade System.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Version', table 'sbhs65.dbo.DesktopModules'; column does not allow nulls. UPDATE fails.
The statement has been terminated.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at DotNetNuke.Data.SqlDatabaseConnectionProvider.ExecuteNonQuery(String connectionString, CommandType commandType, Int32 commandTimeout, String query)
   at DotNetNuke.Data.SqlDataProvider.ExecuteScriptInternal(String connectionString, String script, Int32 timeoutSec)
ClientConnectionId:35d2e1fa-d32e-4f60-a65d-4eb274578da0
Error Number:515,State:2,Class:16

/*  Increase Version column in DesktopModules table to support revision number  */
/********************************************************************************/

ALTER Table dbo.[DesktopModules] 
	ALTER Column [Version] NVARCHAR(50) NOT NULL


09/24/2020 18:13:22 [ERROR] DotNetNuke.Services.Upgrade.Upgrade System.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Version', table 'sbhs65.dbo.DesktopModules'; column does not allow nulls. UPDATE fails.
The statement has been terminated.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at DotNetNuke.Data.SqlDatabaseConnectionProvider.ExecuteNonQuery(String connectionString, CommandType commandType, Int32 commandTimeout, String query)
   at DotNetNuke.Data.SqlDataProvider.ExecuteScriptInternal(String connectionString, String script, Int32 timeoutSec)
ClientConnectionId:35d2e1fa-d32e-4f60-a65d-4eb274578da0
Error Number:515,State:2,Class:16

/*  Increase Version column in DesktopModules table to support revision number  */
/********************************************************************************/

ALTER Table dbo.[DesktopModules] 
	ALTER Column [Version] NVARCHAR(50) NOT NULL


There were other errors:

00:00:01.893 -   Installing Package File Dnn.EditBar.UI_09.07.02_Install:  Success
00:00:02.298 -   Installing Package File DDRMenu_09.07.02_Install:  Error!
00:00:02.569 -   Installing Package File Dnn.PersonaBar.UI_09.07.02_Install:  Success
00:00:03.131 -   Installing Package File DNNCE_AzureConnector_09.07.02_Install:  Success
00:00:03.441 -   Installing Package File DNNCE_Console_09.07.02_Install:  Error!
00:00:03.625 -   Installing Package File DNNCE_CoreMessaging_09.07.02_Install:  Error!
00:00:03.800 -   Installing Package File DNNCE_HtmlEditorManager_09.07.02_Install:  Error!
00:00:03.970 -   Installing Package File DNNCE_Journal_09.07.02_Install:  Error!
00:00:04.144 -   Installing Package File DNNCE_MemberDirectory_09.07.02_Install:  Error!
00:00:04.317 -   Installing Package File DNNCE_ModuleCreator_09.07.02_Install:  Error!
00:00:04.491 -   Installing Package File DNNCE_RazorHost_09.07.02_Install:  Error!
00:00:04.738 -   Installing Package File DNNCE_SiteExportImport_09.07.02_Install:  Success
00:00:04.949 -   Installing Package File DNNCE_SocialGroups_09.07.02_Install:  Error!
00:00:05.119 -   Installing Package File GoogleAnalyticsConnector_09.07.02_Install:  Success
00:00:05.274 -   Installing Package File Telerik_09.07.02_Install:  Success
00:00:06.620 -   Installing Package File jQuery_03.05.01_Install:  Success
00:00:06.794 -   Installing Package File Skin_Xcillion_09.07.02_Install:  Success
00:00:07.194 -   Installing Package File Knockout_03.03.00_Install:  Success
00:00:07.356 -   Installing Package File Selectize_00.12.06_Install:  Success
00:00:07.510 -   Installing Package File AspNetClientCapabilityProvider_09.07.02_Install:  Success
00:00:07.687 -   Installing Package File DNN_SimpleWebFarmCachingProvider_09.07.02_Install:  Success
00:00:07.842 -   Installing Package File DNNCE_FolderProviders_09.07.02_Install:  Success
00:00:08.070 -   Installing Package File DNNConnect.CKEditorProvider_01.01.00_Install:  Success
00:00:11.474 -   Installing Package File Dnn.PersonaBar.Extensions_09.07.02_Install:  Success
00:00:12.826 -   Installing Package File DNNCE_DigitalAssetsManagement_09.07.02_Install:  Error!
00:00:13.015 -   Installing Package File DNNCE_HTML_09.07.02_Install:  Error!
00:00:13.238 -   Installing Package File jQueryMigrate_03.02.00_Install:  Success

There in nothing in the log file that seems to be related to these errors, but I assume that these are related to the SQL errors ...

**Comment by bdukes** [comment]
It looks like your `DesktopModules` table has a row with a `NULL` in the `Version` column, and that's causing a SQL error.  That potentially causes other scripts the fail and results in the errors that you're seeing.  So, I would suggest fixing any `NULL` values in the `Version` column, and see how much farther you get.

**Reply by rstordeur** [reply]
Just did the upgrade from 9.3.2 to 9.8 on our development server and got these exact errors. After reading your answer went into the database and found 2 modules with NULL values (one was a module I developed). Assigned a version number then restored the entire installation and ran the upgrade again. Everything went perfectly. The initial SQL script passed and all of the DNNCE modules upgrades passed as well.

Still functionally checking the install, but thanks a bunch.
